rs/execution_environment/src/scheduler/round_schedule.rs

```diff
@@ -229,4 +231,77 @@ impl RoundSchedule {
 
         (canisters_partitioned_by_cores, canisters)
     }
+
+    /// Finishes canister execution on thread.
+    pub fn finish_canister_execution_on_thread(
+        canister: &mut CanisterState,
+        executed_priority_ids: &mut BTreeMap<CanisterId, AccumulatedPriority>,
+        round_id: ExecutionRound,
+        is_first_iteration: bool,
+        rank: usize,
+    ) {
+        let full_message_execution = match canister.next_execution() {
+            NextExecution::None => true,
+            NextExecution::StartNew => false,
+            // We just finished a full slice of executions.
+            NextExecution::ContinueLong | NextExecution::ContinueInstallCode => true,
+        };
+        let scheduled_first = is_first_iteration && rank == 0;
+
+        if full_message_execution || scheduled_first {
+            // The very first canister is considered to have a full execution round for
+            // scheduling purposes even if it did not complete within the round.
+            canister.scheduler_state.last_full_execution_round = round_id;
+
+            // For now the priority for full execution is always 100.
+            // In future we might use a proportion to the actual load.
+            executed_priority_ids.insert(canister.canister_id(), 100.into());
+        }
+    }
+
+    /// Finishes inner round execution.
+    pub(crate) fn finish_inner_round(
+        &self,
+        canister_states: &mut BTreeMap<CanisterId, CanisterState>,
+        executed_priority_ids: BTreeMap<CanisterId, AccumulatedPriority>,
+    ) {
+        let scheduler_cores = self.scheduler_cores;
+        let compute_capacity_percent =
+            SchedulerImpl::compute_capacity_percent(scheduler_cores) as i64;
+        let number_of_canisters = canister_states.len();
+        let multiplier = (scheduler_cores * number_of_canisters).max(1) as i64;
+
+        // Charge canisters for the priority executed in the previous round.
+        let mut total_executed_priority = 0;
```

Comment:
As above, I have a hard time understanding what "executed priority" is supposed to mean. "Charged priority" or "priority credit / debit" (unsure anymore which means what) would be clearer, IMHO.

Reply:
Renamed to `charged_priority`.